source/extensions/filters/http/cache/cache_policy.h

```diff
@@ -23,82 +22,96 @@ struct CacheEntryUsability {
+// but can also be used to implement deployment specific cache policies.
+//
+// NOT YET IMPLEMENTED: To make CacheFilter use a custom cache policy, store a mutable CachePolicy
+// in FilterState before CacheFilter::decodeHeaders is called.
+class CachePolicy : public Envoy::StreamInfo::FilterState::Object {
 public:
+  // For use in FilterState.
+  static constexpr absl::string_view Name =
+      "io.envoyproxy.extensions.filters.http.cache.cache_policy";
+
   virtual ~CachePolicy() = default;
 
   /**
    * Calculates the lookup key for storing the entry in the cache.
    * @param request_headers - headers from the request the CacheFilter is currently processing.
    */
-  virtual Key createCacheKey(const Http::RequestHeaderMap& request_headers) PURE;
+  virtual Key cacheKey(const Http::RequestHeaderMap& request_headers) PURE;
 
   /**
-   * Determines the cacheability of the response during decoding.
+   * Determines whether the request is eligible for serving from cache and/or having its response
+   * stored in cache.
    * @param request_headers - headers from the request the CacheFilter is currently processing.
-   * @param request_cache_control - the result of parsing the request's Cache-Control header, parsed
-   * by the caller.
-   * @return true if the response may be cached, based on the contents of the request.
+   * @return an enum indicating whether the request is eligible for serving from cache and/or having
+   * its response stored in cache.
    */
-  virtual bool requestCacheable(const Http::RequestHeaderMap& request_headers,
-                                const RequestCacheControl& request_cache_control) PURE;
+  virtual RequestCacheability
+  requestCacheability(const Http::RequestHeaderMap& request_headers) PURE;
 
   /**
    * Determines the cacheability of the response during encoding.
    * @param request_headers - headers from the request the CacheFilter is currently processing.
    * @param response_headers - headers from the upstream response the CacheFilter is currently
    * processing.
-   * @param response_cache_control - the result of parsing the response's Cache-Control header,
-   * parsed by the caller.
-   * @param vary_allow_list - list of headers that the cache will respect when creating the Key for
-   * Vary-differentiated responses.
-   * @return true if the response may be cached.
+   * @return an enum indicating how the response should be handled.
    */
-  virtual bool responseCacheable(const Http::RequestHeaderMap& request_headers,
-                                 const Http::ResponseHeaderMap& response_headers,
-                                 const ResponseCacheControl& response_cache_control,
-                                 const VaryHeader& vary_allow_list) PURE;
+  virtual ResponseCacheability
+  responseCacheability(const Envoy::Http::RequestHeaderMap& request_headers,
+                       const Envoy::Http::ResponseHeaderMap& response_headers) PURE;
```

Comment:
What's the reason for removing vary_allow_list? Is is that rejections based on that make it so we don't even need to query policy?

Reply:
If a response has a `vary` header that isn't in the vary allowlist, there's no need for CacheFilter to ask CachePolicy about it--CacheFilter already has enough information to know not to store it. Thus, there's no need to pass the allowlist to this function.